pkg/expr/nodes.go

```diff
@@ -159,7 +160,7 @@ func (s *Service) buildDSNode(dp *simple.DirectedGraph, rn *rawNode, req *Reques
 
 	dsNode := &DSNode{
 		baseNode: baseNode{
-			id:    dp.NewNode().ID(),
+			id:    rn.idx,
```

Comment:
1. Can please you explain this change? I see that it may have something to do with the change of sorting but I can't understand why that needs to be done. 
2. we can remove `dp` from the method arguments

Reply:
Will add code comment. In particular, the Cloud Watch DS has queries that depend on each other, and for that to work they need to be submitted in order to the datasource.